### PR TITLE
docs: update README.md to align with the style in the README

### PR DESCRIPTION
**Pull request by gitressa**
## The Issue

It says in the README: 

> 12. Update the `README.md` header in Title Case format, for example, use `# DDEV Redis`, not `# ddev-redis`.

the README itself should follow that advice as well :)

## How This PR Solves The Issue

Updates the title of the header to use Title Case.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
 * [Getting started](#getting-started)
 * [How to debug in Github Actions](./README_DEBUG.md)
 
-## What is ddev-addon-template?
+## What is DDEV add-on template?
 
 This repository is a template for providing [DDEV](https://ddev.readthedocs.io) add-ons and services.
 
-In DDEV addons can be installed from the command line using the `ddev add-on get` command (or `ddev get` for versions of DDEV prior to v1.23.5), for example, `ddev add-on get ddev/ddev-redis` or `ddev add-on get ddev/ddev-solr`.
+In DDEV, add-ons can be installed from the command line using the `ddev add-on get` command (or `ddev get` for versions of DDEV prior to v1.23.5), for example, `ddev add-on get ddev/ddev-redis` or `ddev add-on get ddev/ddev-solr`.
```
Comment (rfay) [inline README.md:14]:
```suggestion
In DDEV, add-ons can be installed from the command line using the `ddev add-on get` command, for example, `ddev add-on get ddev/ddev-redis` or `ddev add-on get ddev/ddev-solr`.
```

**Review by rfay — APPROVED** [review]
Looks great, just would probably leave behind the v1.23.4 old `ddev get` since we're touching it.

**Comment by rfay** [comment]
> Update the README.md header in Title Case format, for example, use # DDEV Redis, not # ddev-redis.

I don't think this is a good idea. Not sure where it came from. I'm afraid "DDEV Redis" is not a thing, but `ddev-redis` is.

**Comment by gitressa** [comment]
It's from the template guide: https://github.com/ddev/ddev-addon-template?tab=readme-ov-file#getting-started.

**Comment by rfay** [comment]
Right. I think you added it, and I wasn't paying adequate attention :)

**Comment by gitressa** [comment]
Heh, that could be! Though in the comment below from you, it sounds like you did evaluate the proposal. But as a Danish prime minister once said, "You have a standpoint, until you take a new one" :)

> I don't see any reason this shouldn't be done, but the best path is for you to do PRs in the various repos, which I think you've already started, thanks.

From https://github.com/ddev/ddev/issues/5641.

**Comment by rfay** [comment]
It's fine in the context of the template, where the DDEV Add-on Template is a "thing", but IMO it doesn't make sense in add-ons.

**Comment by gitressa** [comment]
I do see a slight advantage in using the machine name in the title, and streamlining the style guide to use that, so that people can then memorize it, for later use on the CLI.

So in other words, it should be the reverse, like this?

> 12. Update the README.md header, adding the machine name of the add-on, for example `# ddev-redis`, not `# DDEV Redis`.

I can create an MR if you confirm.

**Comment by rfay** [comment]
I do prefer that, thanks. You can do it in this PR if you want.

**Comment by gitressa** [comment]
Great, the MR is now updated. I reversed the order to Header first and then Description, since that's usually the workflow.

**Review by rfay — APPROVED** [review]
Works for me, thanks.

**Pull request merged**